### DISCUSSION TITLE
Pyodide node in VsCode extension

### DISCUSSION THREAD

**Discussion by jagalindo** [pyodide/pyodide#4253] [Q&A]
Hi all, I'm trying to develop a vscode extension that internally rely on the pyodide node version (https://github.com/flamapy/vscode_extension.). My integration is properly working when I launch a debuggin sesion, and everything works as expected. However, when I pacakge the extension and install it I get this error

TypeError: Invalid host defined options
	at initNodeModules (/Users/jagalindo/.vscode/extensions/diversolab.flamapy-0.0.8/node_modules/pyodide/pyodide.js:9:802)
	at loadPyodide (/Users/jagalindo/.vscode/extensions/diversolab.flamapy-0.0.8/node_modules/pyodide/pyodide.js:9:9398)
	at getPyodideInstance (/Users/jagalindo/.vscode/extensions/diversolab.flamapy-0.0.8/extension.js:22:39)

Any clues of what it can be?

**Comment by hoodmane** [comment]
Which node version?

**Reply by jagalindo** [reply]
In my computer I have the version 21 installed. I setted the minimum version for the extension to 1.82.0 code version which came with electron 25.8.0 and node 18.15.0. Therefore, I'm assuming it rely on the 18.15 version

**Reply by hoodmane** [reply]
Could you try running:
```js
const nodeUrlMod = await import("url");
const nodeFsPromisesMod = await import("fs/promises");
const nodeVmMod = await import("vm");
const nodePath = await import("path");
const fs = await import("fs");
const crypto = await import("crypto");
const ws = await import("ws");
const child_process = await import("child_process");
```
and seeing if any of these imports fail? Maybe some modules are just not available.

**Comment by jagalindo** [comment]
Thank you a lot for pointing to the loading of dependencies. There is something, sort of related. It does seems that electron is not allowing mjs modules until its version 28. Therefore, I could just wait until it supports it and eventually it will work.
